File: Java/Program/Shirt.java
/*Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 2 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

public class Shirt extends Clothing{
    
    /*Attribute */
    private String color;
    private String sleeve_type;

    /*Constructor */
    public Shirt(){
        this.color = "";
        this.sleeve_type = "";
    }

    public Shirt(String idProduct, String name, String brand, String price, String size, String material, String gender, String color, String sleeve_type) {
        setIdProduct(idProduct);
        setName(name);
        setBrand(brand);
        setPrice(price);
        setSize(size);
        setMaterial(material);
        setGender(gender);
        this.color = color;
        this.sleeve_type = sleeve_type;
    }

    /*Getter & Setter */
    // Getter
    public String getColor() {
        return color;
    }

    public String getSleeve_type() {
        return sleeve_type;
    }

    // Setter
    public void setColor(String color) {
        this.color = color;
    }

    public void setSleeve_type(String sleeve_type) {
        this.sleeve_type = sleeve_type;
    }
}

File: Cpp/Program/Shirt.cpp
#include <iostream>
#include <string>
#include "Clothing.cpp"

using namespace std;

class Shirt : public Clothing
{
    /*Attribute*/
    private:
        string color;
        string sleeve_type;

    /*Constuctor*/
    public:
        Shirt(){

        }

        Shirt(string idProduct, string name, string brand, string price, 
                string size, string material, string gender, string color, 
                string sleeve_type) : Clothing(idProduct, name, brand, price, size, material, gender){
            this->color = color;
            this->sleeve_type = sleeve_type;
        }

    /*Getter & Setter*/
    // Getters
    string getColor() {
        return this->color;
    }

    string getSleeveType() {
        return this->sleeve_type;
    }

    // Setter
    void setColor(string color) {
        this->color = color;
    }

    void setSleeveType(string sleeve_type) {
        this->sleeve_type = sleeve_type;
    }

    ~Shirt()
    {}
};
